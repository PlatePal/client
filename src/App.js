import { Route, Switch, BrowserRouter } from 'react-router-dom';

import DietaryRestrictions from './Containers/Pages/DietaryRestrictions/DietaryRestrictions.js';
import LoginPage from './Containers/Pages/Login/Login';
import LandingPage from './Containers/Pages/LandingPage/LandingPage';
import MetricPage from './Containers/Pages/Metrics/Metrics';
import MainRecipeModal from './Containers/Pages/MainRecipeModal/MainRecipeModal';
import ExpandedSavedRecipe from './Containers/Pages/ExpandedSavedRecipe/ExpandedSavedRecipe.js';
import ExpandedRecipe from './Containers/Pages/ExpandedRecipe/ExpandedRecipe.js';
import IngredientSelection from './Containers/Pages/IngredientSelection/IngredientSelection.js';
import ProfilePage from './Containers/Pages/ProfilePage/ProfilePage.js';

import { ThemeProvider } from '@material-ui/core';
import mainUITheme from './assets/styles/mainUITheme';
import 'fontsource-roboto';

import routes from './constant/routes';
import Navbar from './component/navbars/Navbar.js';

function App() {

  return (
    <BrowserRouter>
      <ThemeProvider theme={mainUITheme}>
        <Switch>
          <Route path={`/${routes.login}`} component={LoginPage} />
          <Route path='/' exact component={LandingPage} />
          <Route path={`/${routes.metrics}`} exact component={MetricPage} />
          <Route
            path={`/${routes.dietaryRestriction}`}
            exact
            component={DietaryRestrictions}
          />
          <Route
            path={`/${routes.expandedSavedRecipe}`}
            exact
            component={ExpandedSavedRecipe}
          />
          <Route
            path={`/${routes.expandedRecipe}`}
            exact
            component={ExpandedRecipe}
          />
          <Route
            path={`/${routes.ingredientSelection}`}
            exact
            component={IngredientSelection}
          />
          <Route
            path={`/${routes.main}`}
            render={(props) => 
              <div>
                <Navbar fill='home'/>
                <MainRecipeModal/>  
              </div>
            }
          />
          <Route 
            path={`/${routes.profile}`} 
            render={(props) => 
              <div>
                <Navbar fill='person'/>
                <ProfilePage/>
              </div>
            }
          /> 
        </Switch>
      </ThemeProvider>
    </BrowserRouter>
  );
}

export default App;
