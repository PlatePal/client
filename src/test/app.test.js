import React from "react";
import { cleanup, render } from "@testing-library/react";
import App from "../App";
import { shallow } from "enzyme";
// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

it("Check App renders using react testing library", () => {
  const { queryByText, getByTestId } = render(<App />);
  expect(queryByText("Login")).toBeTruthy;
  expect(queryByText("Register")).toBeTruthy;
  expect(getByTestId("logo")).toBeTruthy;
});

test("Check App renders using enzyme", () => {
  const appWrapper = shallow(<App />);

  expect(appWrapper.getElement().props.children.type).toEqual("div");
});
