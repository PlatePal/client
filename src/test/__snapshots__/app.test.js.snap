// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fullPage"
>
  <img
    alt="plate pal logo"
    className="logoImage"
    src="plate_pal_logo.png"
  />
  <h3
    className="logo"
    data-testid="logo"
  >
    PlatePal
  </h3>
  <button
    className="loginButton"
    onClick={[Function]}
  >
    Login
  </button>
  <button
    className="registerButton"
  >
    Register
  </button>
</div>
`;
